### PR TITLE
Clean up / add tests for new dataset versioning

### PR DESCRIPTION
**Pull request by convoliution**
There's some reorganization of tests that were previously written, sorry!
Double-sorry for the diff being less nice than I expected.

Covers:

- Dataset, Path DatasetVersion, S3 DatasetVersion
- create and get
- metadata CRUD
- log to and get from ExperimentRun

Still need to adjust/remove the tests for the old dataset versioning system.

### CODE REVIEW

File: client/verta/tests/test_dataset_versioning/test_dataset_version.py [hunk 1]
```diff
@@ -32,69 +102,69 @@ def test_create_get_s3(self, client, created_datasets):
...
+        version2 = dataset.create_version(S3("s3://verta-starter/census-train.csv"))
+
+        experiment_run.log_dataset_version(key1, version1)
+        experiment_run.log_dataset_version(key2, version2)
+
+        for key, version in [(key1, version1), (key2, version2)]:
+            retrieved_version = experiment_run.get_dataset_version(key)
+            assert retrieved_version.id == version.id
+
+            retreved_components = sorted(retrieved_version.get_content().list_components(), key=lambda component: component.path)
```
Comment (evgeniilezhnin) [inline client/verta/tests/test_dataset_versioning/test_dataset_version.py]:
```suggestion
            retrieved_components = sorted(retrieved_version.get_content().list_components(), key=lambda component: component.path)
```

File: client/verta/tests/test_dataset_versioning/test_dataset_version.py [hunk 2]
```diff
@@ -32,69 +102,69 @@ def test_create_get_s3(self, client, created_datasets):
...
+            components          = sorted(version.get_content().list_components(),           key=lambda component: component.path)
+            assert retreved_components == components
```
Comment (evgeniilezhnin) [inline client/verta/tests/test_dataset_versioning/test_dataset_version.py]:
```suggestion
            assert retrieved_components == components
```

**Review by evgeniilezhnin — APPROVED** [review]

File: client/verta/tests/test_dataset_versioning/test_dataset_version.py (hunk 1 above)
Comment (convoliution) [reply client/verta/tests/test_dataset_versioning/test_dataset_version.py]: Thanks Evgenii 😅

**Comment by convoliution** [comment]
I'm closing this in favor of https://github.com/VertaAI/modeldb/pull/1704, because there's too many merge conflicts that could be resolved with a single rebase.

**Pull request closed without merging**